### PR TITLE
fixed categories save bug in contacts

### PR DESCRIPTION
**Pull request by marcelmoosbrugger**
| Q | A
| --- | ---
| Bug fix? | yes
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes https://github.com/sulu/sulu-standard/issues/671
| Related issues/PRs | https://github.com/massiveart/husky/pull/692
| License | MIT
| Documentation PR | none

#### What's in this PR?

In the frontend an issue with the convertion form categories id to categories is made. In the backend the serialization context 'partialCategories' was added to accounts.

### CODE REVIEW

File: src/Sulu/Bundle/ContactBundle/Controller/AccountController.php [hunk 1]
```diff
@@ -578,7 +578,7 @@ public function putAction($id, Request $request)
 
                 $view = $this->view($acc, 200);
                 $view->setSerializationContext(
-                    SerializationContext::create()->setGroups(['fullAccount', 'partialContact', 'partialMedia'])
+                    SerializationContext::create()->setGroups(['fullAccount', 'partialContact', 'partialMedia', 'partialCategory'])
```
Comment (danrot) [inline src/Sulu/Bundle/ContactBundle/Controller/AccountController.php]: Don't you think that should be extracted into an own variable as in the `ContactController`?

Comment (danrot) [inline src/Sulu/Bundle/ContactBundle/Controller/AccountController.php]: You forgot it also in the `postAction` 😉

File: src/Sulu/Bundle/ContactBundle/Resources/public/js/services/account-manager.js [hunk 2]
```diff
@@ -67,8 +67,10 @@ define([
 
             if (!!data.categories) {
                 account.get('categories').reset();
-                Util.foreach(data.categories, function(categoryId) {
-                    var category = Category.findOrCreate({id: categoryId});
+                Util.foreach(data.categories, function(category) {
+                    if (Util.typeOf(category) === 'number') {
```
Comment (danrot) [inline src/Sulu/Bundle/ContactBundle/Resources/public/js/services/account-manager.js]: Why is this necessary? Shouldn't it be always the same type?

File: src/Sulu/Bundle/ContactBundle/Resources/public/js/services/contact-manager.js [hunk 3]
```diff
@@ -63,8 +63,10 @@ define([
 
             if (!!data.categories) {
                 contact.get('categories').reset();
-                Util.foreach(data.categories, function(id) {
-                    var category = Category.findOrCreate({id: id});
+                Util.foreach(data.categories, function(category) {
+                    if (Util.typeOf(category) === 'number') {
```
Comment (danrot) [inline src/Sulu/Bundle/ContactBundle/Resources/public/js/services/contact-manager.js]: Same as above

**Comment by marcelmoosbrugger** [comment]
@danrot The issues you mentioned in your comments are fixed :) Note that a added a little PR from husky (just a little style issue I noticed in the contacts).

File: src/Sulu/Bundle/CategoryBundle/Resources/public/js/validation/types/categoryList.js [hunk 4]
```diff
@@ -38,8 +38,11 @@ define([
                     App.emit('sulu.content.changed');
                 },
 
-                setValue: function(value) {
-                    App.dom.data($el, 'selected', value);
+                setValue: function(categories) {
+                    categories = categories.map(function(category) {
+                        return (!!category.id) ? category.id : category;
```
Comment (danrot) [inline src/Sulu/Bundle/CategoryBundle/Resources/public/js/validation/types/categoryList.js]: Same question as before (I think you only moved the problem to somewhere else): Why do you get different values here?
